### PR TITLE
JBPM-5710 KIE Persistence Cleanup

### PR DESCRIPTION
**Pull request by marianbuenosayres**
This set of PRs is a set of modifications to allow multiple persistences without all of them depending on JPA:

- Creating drools-persistence-api and jbpm-persistence-api
- SingleSessionCommandService refactors
- PersistenceContext and ProcessPersistenceContext refactors:
- TaskPersistenceContext refactors
- Changes in the current persistence model interfaces
- TimerJobFactoryManager / TimerService refactors

### CODE REVIEW

**Comment by mswiderski** [comment]
@marianbuenosayres could you please run locally build with -Dfull as there seems to be some duplicated classes/dependencies and that's why build failed.

**Comment by marianbuenosayres** [comment]
Thank you @mswiderski . I pushed an update to the drools-persistence-api POM file that seems to solve the issue

File: drools-persistence/drools-persistence-api/src/main/java/org/drools/persistence/PersistentSession.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package org.drools.persistence;
```
Comment (mswiderski) [inline drools-persistence/drools-persistence-api/src/main/java/org/drools/persistence/PersistentSession.java]: missing license header and maybe small java doc would be nice too

File: drools-persistence/drools-persistence-api/src/main/java/org/drools/persistence/PersistentWorkItem.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package org.drools.persistence;
```
Comment (mswiderski) [inline drools-persistence/drools-persistence-api/src/main/java/org/drools/persistence/PersistentWorkItem.java]: same here  - missing license header and maybe small java doc would be nice too

File: drools-persistence/drools-persistence-api/src/main/java/org/drools/persistence/processinstance/InternalWorkItemManager.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package org.drools.persistence.processinstance;
```
Comment (mswiderski) [inline drools-persistence/drools-persistence-api/src/main/java/org/drools/persistence/processinstance/InternalWorkItemManager.java:17]: missing license header and maybe small java doc would be nice too

File: drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/info/SessionInfo.java [hunk 4]
```diff
@@ -26,12 +27,13 @@
...
 import javax.persistence.Version;
 
+import org.drools.persistence.PersistentSession;
 import org.drools.persistence.SessionMarshallingHelper;
 import org.drools.persistence.Transformable;
 
 @Entity
 @SequenceGenerator(name="sessionInfoIdSeq", sequenceName="SESSIONINFO_ID_SEQ")
-public class SessionInfo implements Transformable {
+public class SessionInfo implements Transformable, PersistentSession {
```
Comment (mswiderski) [inline drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/info/SessionInfo.java]: Transformable seems to be redundant here as PersistentSession already extents it

File: drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/info/WorkItemInfo.java [hunk 5]
```diff
@@ -36,14 +37,15 @@
...
+import org.drools.persistence.PersistentWorkItem;
 import org.drools.persistence.Transformable;
 import org.kie.api.runtime.Environment;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 @Entity
 @SequenceGenerator(name="workItemInfoIdSeq", sequenceName="WORKITEMINFO_ID_SEQ")
-public class WorkItemInfo implements Transformable {
+public class WorkItemInfo implements Transformable, PersistentWorkItem {
```
Comment (mswiderski) [inline drools-persistence/drools-persistence-jpa/src/main/java/org/drools/persistence/info/WorkItemInfo.java]: same comment regarding Transformable being redundant

**Review by mswiderski — CHANGES_REQUESTED** [review]
some small comments to adjust, but aver all looks good

@mariofusco do you mind taking a extra look at the TimerJobFactoryType?

**Review by mswiderski — APPROVED** [review]

**Pull request merged**